src/libs/classes.ts

```diff
@@ -30,17 +27,13 @@ async function upsertClass(db: Db, user: string, scheduleClass: {
 	type?: ClassType,
 	teacher: Teacher
 }) {
-	// Input validation best validation
-	if (typeof db !== 'object') { throw new Error('Invalid database connection!'); }
+	// Set default
 	if (typeof scheduleClass._id !== 'string') { scheduleClass._id = ''; }
 
-	if (typeof user               !== 'string') { throw new Error('Invalid username!'); }
-	if (typeof scheduleClass      !== 'object') { throw new Error('Invalid class object!'); }
-	if (typeof scheduleClass.name !== 'string') { throw new Error('Invalid class name!'); }
+	// If no block or type, default to 'other'
+	if (!scheduleClass.block) { scheduleClass.block = Block.OTHER; }
+	if (!scheduleClass.type)  { scheduleClass.type  = ClassType.OTHER; }
```

Comment:
I don't think we can use null coalescing, and unless we started doing destructuring in the parameter, I don't think default arguments would work here, since they're object properties.

Reply:
We might be able to do

```typescript
scheduleClass.block = scheduleClass.block ?? Block.OTHER
```

unless I'm missing something, but whatever seems cleaner to you